os/kernel/pthread/pthread_create.c

```diff
@@ -260,6 +260,10 @@ int pthread_create(FAR pthread_t *thread, FAR const pthread_attr_t *attr, pthrea
 
 	trace_begin(TTRACE_TAG_TASK, "pthread_create");
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	ASSERT((sched_self()->flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL);
```

Comment:
How about moving this before `trace_begin`?
Because we don't need to start `trace` if we will not create the pthread.

Reply:
Done